### DISCUSSION TITLE
Reading UnityCapture virtual camera frames with OpenCV

### DISCUSSION THREAD

**Discussion by dohrmannc** [letmaik/pyvirtualcam#112] [Q&A]
Hello,

First of all, thank you for the library, it's really awesome. I have a use case that I haven't been able to find a solution for.
 
I have 2 webcams and I am trying to set up a virtual webcam for each in order to run a variety of analysis scripts on the frames from both. The intent of using virtual cameras is to avoid blocking the camera by connecting to it directly with OpenCV, or having to broadcast the frames to different processes through some other means.

I'm on Windows, so it looks like my only option for a backend that supports multiple virtual cameras is UnityCapture. 

I have been able to read the OBS virtual camera like so:
```python
# sender
with pyvirtualcam.Camera(
    ...
) as virtual_camera:
    while not self.stop_event.is_set():
        _, frame = camera.read()
        virtual_camera.send(frame)
```

```python
# receiver
camera = cv2.VideoCapture(3, cv2.CAP_DSHOW)
_, frame = camera.read()
do_some_processing(frame)
```

The same code does not work for the UnityCapture virtual cameras. I can see that they are correctly receiving frames by opening them as sources in OBS. 

I get the following error when trying to grab frames with the above snippet:

```
global cap.cpp:344 cv::VideoCapture::open VIDEOIO(DSHOW): backend is generally available but can't be used to capture by index
```

I've tried experimenting with different values from your API docs but haven't been able to solve it.

Do you have any suggestions?

**Comment by GermanWarez** [comment]
The error is related to opencv, read e.g. https://forum.opencv.org/t/running-on-windows-10-cant-open-camera-by-index-on-windows-11/14432
I'd try the fixes recommended in that post, and otherwise ask for support at opencv.

You could try to open both cameras with e.g. VLC or any other app that can display video streams - this should work, assuming there's no error in pyvirtualcam/unitycam.
